client-java/controller/src/main/java/org/evomaster/client/java/controller/problem/rpc/JavaXConstraintHandler.java

```diff
@@ -20,20 +20,24 @@
      */
     public static void handleParam(NamedTypedValue namedTypedValue, Annotation annotation){
         Class<?> cons = annotation.annotationType();
-        if (cons.getSimpleName().equals("NotNull")){
-            namedTypedValue.setNullable(false);
+
+        JavaXConstraintSupportType supportType = JavaXConstraintSupportType.getSupportType(cons.getSimpleName());
+        if (supportType == null){
+            SimpleLogger.error("ERROR: Not handle constraints with a specified annotation:"+ cons.getName());
             return;
         }
 
         boolean solved = false;
-        if (namedTypedValue instanceof PrimitiveOrWrapperParam){
-            solved = handlePrimitiveOrWrapperParam((PrimitiveOrWrapperParam) namedTypedValue, annotation);
-        } else if (namedTypedValue instanceof StringParam){
-            solved = handleStringParam((StringParam) namedTypedValue, annotation);
-        } else if (namedTypedValue instanceof CollectionParam){
-            solved = handleCollection((CollectionParam) namedTypedValue, annotation);
-        }  else if (namedTypedValue instanceof MapParam){
-            solved = handleMapParam((MapParam) namedTypedValue, annotation);
+        switch (supportType){
+            case NOT_NULL: solved = handleNotNull(namedTypedValue); break;
+            case NOT_EMPTY: solved = handleNotEmpty(namedTypedValue); break;
+            case NOT_BLANK: solved = handleNotBlank(namedTypedValue); break;
+            case SIZE: solved = handleSize(namedTypedValue, annotation); break;
+            case PATTERN: solved = handlePattern(namedTypedValue, annotation); break;
+            case DECIMAL_MAX:
+            case MAX: solved = handleMax(namedTypedValue, annotation, supportType); break;
+            case DECIMAL_MIN:
+            case MIN: solved = handleMin(namedTypedValue, annotation, supportType); break;
```

Comment:
are these all supported cases?
should we have a `default` throwing an exception if reached?

Reply:
or log it?

Reply:
it is handled at line 26 by checking whether we support the annotation with its simple name. 
To be safe, I will add a default to have a further check.